cou/steps/hypervisor.py

```diff
@@ -228,7 +228,13 @@ def _generate_upgrade_steps(
 
             units = group.app_units[app.name]
             logger.info("generating upgrade steps for %s units of %s app", app.name, units)
-            steps.extend(app.upgrade_steps(target, units, force))
+
+            if app.charm == "nova-compute":
+                # NOTE(gabrielcocenza) it's important to disable the scheduler before upgrading
+                # any application, that is why we insert in the first position.
+                steps = app.upgrade_steps(target, units, force) + steps
+            else:
+                steps.extend(app.upgrade_steps(target, units, force))
```

Comment:
I think that this is not needed, since all apps will be in upgraded order, where nova-compute is first. If not I think we should rather make sure that the order of apps is correct even for pre- and post-upgrade steps.

Reply:
Nova-compute might not be the first when colocated with another OpenStack service. E.g: cinder is up into the upgrade order and will be the first application to enter the data-plane group

Reply:
Ok, in tht case could we re-order all apps and then pass them to the HypervosorUpgradePlaner?

It will be much cleaner then these if statments. Would that be ok?

Now I realized, if cinder should be upgraded first, it's ok that we change that here for nova-compute to be first? Maybe we can simply move disable to pre-upgrade and enable to post-upgrade (we already did that).